### PR TITLE
Migration/vibratobracket

### PR DESCRIPTION
**Pull request by rvilarl**
vibratobracket!

### CODE REVIEW

File: src/vibratobracket.ts [hunk 1]
```diff
@@ -9,21 +9,45 @@
 import { Vex } from './vex';
 import { Element } from './element';
 import { Vibrato } from './vibrato';
+import { Note } from './note';
+
+export interface VibratoBracketData {
```
Comment (0xfe) [inline src/vibratobracket.ts:14]: This can be inline.

File: src/vibratobracket.ts [hunk 2]
```diff
@@ -9,21 +9,45 @@
...
+  stop: Note;
+  start: Note;
+}
+
+export interface VibratoBracketRenderOptions {
```
Comment (0xfe) [inline src/vibratobracket.ts:19]: This can also be inline.

File: src/vibratobracket.ts [hunk 3]
```diff
@@ -37,37 +61,40 @@ export class VibratoBracket extends Element {
...
 
-    const y = this.start
-      ? this.start.getStave().getYForTopText(this.line)
-      : this.stop.getStave().getYForTopText(this.line);
+    const y = startStave ? startStave.getYForTopText(this.line) : stopStave ? stopStave.getYForTopText(this.line) : 0;
 
     // If start note is not set then vibrato will be drawn
     // from the beginning of the stave
-    const start_x = this.start ? this.start.getAbsoluteX() : this.stop.getStave().getTieStartX();
+    const start_x = this.start ? this.start.getAbsoluteX() : stopStave ? stopStave.getTieStartX() : 0;
```
Comment (0xfe) [inline src/vibratobracket.ts:89]: This seems to change the logic to silently fail. If `this.start` is not set, then `stopStave` must be set.

Comment (rvilarl) [reply src/vibratobracket.ts:89]: changed to assertion

File: src/vibratobracket.ts (hunk 2 above)
Comment (rvilarl) [reply src/vibratobracket.ts]: done

File: src/vibratobracket.ts (hunk 1 above)
Comment (rvilarl) [reply src/vibratobracket.ts]: done

**Comment by rvilarl** [comment]
Ready

**Comment by 0xfe** [comment]
Looks good.

**Pull request merged**